test/e2e/create_test.go

```
@@ -107,13 +107,21 @@ func (tc *testContext) deleteMachineSet(ms *mapi.MachineSet) error {
 }
 
 // waitForWindowsMachines waits for a certain amount of Windows Machines to reach a certain phase
+// if machineCount = 0, it implies we are only waiting for Machines to be deleted and the phase is
+// ignored in this case.
 func (tc *testContext) waitForWindowsMachines(machineCount int, phase string) error {
-	machineCreationTimeLimit := time.Minute * 5
+	machineStateTimeLimit := time.Minute * 5
```

Please return an error if `machineCount == 0 && phase != ""`